ftest/src/test/java/org/arquillian/reporter/GreeterVerifier.java

```diff
@@ -0,0 +1,160 @@
+import static org.arquillian.reporter.impl.asserts.TestMethodReportAssert.assertThatTestMethodReport;
+import static org.arquillian.reporter.impl.asserts.TestSuiteReportAssert.assertThatTestSuiteReport;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class GreeterVerifier {
+
+    private static File reportInJson;
+    private static ExecutionReport executionReport;
+
+    @BeforeClass
+    public static void setUp() throws FileNotFoundException {
+        JUnitCore.runClasses(GreeterTest.class);
+        reportInJson = new File("target/report.json");
+        executionReport = ReportJsonParser.parse("target/report.json");
+    }
+
+    @Test
+    public void verify_test_report_creation_in_json() throws FileNotFoundException {
+        assertThat(reportInJson).exists();
+    }
+
+    @Test
+    public void verify_execution_report() {
+        assertThatExecutionReport(executionReport)
+            .hasName("execution")
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(0);
+    }
+
+    @Test
+    public void verify_test_suite_report() {
+        List<TestSuiteReport> testSuiteReports = executionReport.getTestSuiteReports();
+        TestSuiteReport testSuiteReport = testSuiteReports.get(0);
+
+        assertThat(testSuiteReports).size().isEqualTo(1);
+
+        assertThatTestSuiteReport(testSuiteReport)
+            .hasName(ArquillianCoreKey.TEST_SUITE_NAME)
+            .hasNumberOfSubReportsAndEntries(0);
+
+        assertThat(testSuiteReport.getConfiguration().getName())
+            .isEqualTo(ReporterCoreKey.GENERAL_TEST_SUITE_CONFIGURATION_REPORT);
+
+        assertThat(testSuiteReport.getTestClassReports()).size().isEqualTo(1);
+    }
+
+    @Test
+    public void verify_test_suite_configuration_report() {
+        ConfigurationReport configuration = executionReport.getTestSuiteReports().get(0).getConfiguration();
+        List<Report> subReports = configuration.getSubReports();
+        Report containerConfigurationSubReport = subReports.get(0);
+        Report environmentConfigurationSubReport = subReports.get(1);
+
+        assertThatReport(configuration)
+            .hasName(ReporterCoreKey.GENERAL_TEST_SUITE_CONFIGURATION_REPORT)
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(2);
+
+        assertThat(containerConfigurationSubReport.getName())
+            .isEqualTo(new UnknownStringKey("containers"));
+        assertThat(containerConfigurationSubReport.getEntries()).size().isEqualTo(0);
+        assertThat(containerConfigurationSubReport.getSubReports()).size().isEqualTo(1);
+
+        assertThat(environmentConfigurationSubReport.getName())
+            .isEqualTo(EnvironmentKey.ENVIRONMENT_SECTION_NAME);
+        assertThat(environmentConfigurationSubReport.getEntries()).size().isEqualTo(8);
```

Comment:
When I went through the code, I haven't seen any verification of the entries; I believe that you can find some pattern that would be common for all environments